pages/blog/posts/advanced-json-schema-with-slashdb.md

```diff
@@ -0,0 +1,749 @@
+            "oneOf": [
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "pattern": "^(?![xX][mM][lL])[a-zA-Z_][a-zA-Z0-9@_~-]*$",
+                        "maxLength": 128
+                    }
+                },
+                {
+                    "type": "null"
+                }
+            ]
+        },
+        "http_methods": {
+            "type": "object",
+            "additionalProperties": false,
+            "minProperties": 1,
+            "properties": {
+                "DELETE": {
+                    "type": "boolean"
+                },
+                "GET": {
+                    "type": "boolean"
+                },
+                "POST": {
+                    "type": "boolean"
+                },
+                "PUT": {
+                    "type": "boolean"
+                }
+            }
+        },
+        "query_id": {
+            "type": "string",
+            "pattern": "^(?![xX][mM][lL])[a-zA-Z_][a-zA-Z0-9@_~-]*$",
+            "maxLength": 128
+        },
+        "read": {
+            "oneOf": [
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "pattern": "^(?![xX][mM][lL])[a-zA-Z_][a-zA-Z0-9@_~-]*$",
+                        "maxLength": 128
+                    }
+                },
+                {
+                    "type": "null"
+                }
+            ]
+        },
+        "sqlstr": {
+            "type": "string",
+            "minLength": 1
+        },
+        "write": {
+            "oneOf": [
+                {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "pattern": "^(?![xX][mM][lL])[a-zA-Z_][a-zA-Z0-9@_~-]*$",
+                        "maxLength": 128
+                    }
+                },
+                {
+                    "type": "null"
+                }
+            ]
+        },
+        "url_template": {
+            "type": [
+                "string",
+                "null"
+            ]
+        }
+    },
+    "required": [
+        "query_id",
+        "database",
+        "http_methods",
+        "sqlstr"
+    ],
+    "optional": [
+        "creator",
+        "desc",
+        "execute",
+        "read",
+        "write",
+        "url_template"
+    ]
+}
+```
+
+We see that the `pattern` attribute is present in several portions of the schema here.  Used in conjunction with `string`, `pattern` allows you to specify a regular expression that a value must match against.
+
+E.g - the `query_id` attribute, which is a unique string that identifies the query, must be no more than 128 characters long (`maxLength` property), and the string must match [this regular expression](https://regex101.com/r/2crKdN/2) - `^(?![xX][mM][lL])[a-zA-Z_][a-zA-Z0-9@_~-]`.
```

Comment:
"e.g." is typically not used at the beginning of a sentence.  I recommend writing it out, e.g. "For example, ...".

Additionally, the em dash doesn't fit here.